posframe.el

```diff
@@ -663,13 +663,16 @@ WIDTH and MIN-WIDTH."
                           frame height min-height width min-width)))
                    posframe height min-height width min-width)))))
 
+(defun posframe-buffer-frame (buf-or-name)
+  "Return the posframe pertaining to BUF-OR-NAME.
+BUF-OR-NAME can be a buffer or a buffer name."
+  (when-let ((buffer (get-buffer buf-or-name)))
+    (with-current-buffer buffer posframe--frame)))
+
 (defun posframe-hide (buf-or-name)
   "Hide posframe pertaining to BUF-OR-NAME."
-  (dolist (frame (frame-list))
-    (let ((buffer-info (frame-parameter frame 'posframe-buffer)))
-      (when (or (equal buf-or-name (car buffer-info))
-                (equal buf-or-name (cdr buffer-info)))
-        (posframe--make-frame-invisible frame)))))
+  (when-let ((frame (posframe-buffer-frame buf-or-name)))
```

Comment:
when posframe is killed, this command will not work

Reply:
Same here, I don't understand what you mean... Why would anyone want to hide a frame that doesn't exist? (And again, this will just return nil in that case, which seems graceful).

Reply:
sorry, i mean posframe buffer, or buf-or-name, your change is required buffer vairable posframe--frame , if buf-or-name is killed, this command will not work

Reply:
Hm, I still don't understand: If the buffer is still alive, the buffer-local variable posframe--frame records the frame we want. If the buffer has been killed already then the frame has already been deleted as well (from kill-buffer-hook).

Reply:
in my opinion, kill-buffer-hook seem to unreliable at some condition.

Reply:
Hm, I don't know about that, but I changed also other things related to this, including the kill-buffer-hook modifications. I have been using this for a few days and it works for me, but my use case is quite limited. Maybe just try it out for a few days or make some tests yourself before merging? (Of course I mean the whole PR, i.e. all the commits together)

Reply:
Best would be to actually have a test suite, but... that's work! :-P

Reply:
i suggest do not change, the current version is very reliable

Reply:
I don't know how reliable it is, but it seems wrong to me, at the very least (unless I'm missing something) it's doing much more than is necessary (which if anything should make it less reliable), as explained in the commit message of 346a1669, so I'm definitely going to continue using my version unless I find any problems or am proven wrong. I guess you could just cherry pick some of the independent commits for the time being, though.

Reply:
I do not remeber the detail, but I know kill-buffer-hook seem to unreliable in some condition, when I kill posframe buffer, posframe's frame will not be killed

In the earlier version of posframe, I use your approach to hide and kill posframe, 
finally, I have to use current version to hide and kill posframe. 

I will merge this PR, but revert this two commit.

Reply:
OK, that's unfortunate, because even if there would be the problems you allude to I'm convinced this is not the way to solve them, and I can't really help solve them in a better way until I know what they are. We could also create some tests for it.
Also, I could have at least rearranged the commits so that my version would be a clean superset of yours (i.e., you wouldn't have to revert). Oh well...
Thank you.